code/modules/antagonists/traitor/objectives/demoralise_graffiti.dm

```diff
@@ -0,0 +1,304 @@
+	fail_objective()
+
+/datum/traitor_objective/demoralise/graffiti/ungenerate_objective()
+	. = ..()
+	if (rune)
+		UnregisterSignal(rune, COMSIG_PARENT_QDELETING)
+		UnregisterSignal(rune, COMSIG_DEMORALISING_EVENT)
+		rune = null
+
+// Extending the existing spraycan item was more trouble than it was worth, I don't want or need this to be able to draw arbitrary shapes.
+/obj/item/traitor_spraycan
+	name = "seditious spraycan"
+	desc = "This spraycan deploys a subversive pattern containing subliminal priming agents over a 3x3 area. Contains enough primer for just one final coating."
+	icon = 'icons/obj/crayons.dmi'
+	icon_state = "deathcan"
+	worn_icon_state = "spraycan"
+	inhand_icon_state = "spraycan"
+	lefthand_file = 'icons/mob/inhands/equipment/hydroponics_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/hydroponics_righthand.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	var/paint_color = "#780000"
+	var/static/list/no_draw_turfs = typecacheof(list(/turf/open/space, /turf/open/openspace, /turf/open/lava, /turf/open/chasm))
+
+	/// Are we currently drawing? Used to prevent spam clicks for do_while
+	var/drawing_rune = FALSE
+	/// Set to true if we finished drawing something, this spraycan is now useless
+	var/expended = FALSE
+
+/obj/item/traitor_spraycan/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if (expended)
+		user.balloon_alert(user, "all out of paint...")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (drawing_rune)
+		user.balloon_alert(user, "already busy!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (!proximity || !check_allowed_items(target) || !isliving(user))
+		return
+
+	if (isturf(target))
+		try_draw_new_rune(user, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (istype(target, /obj/effect/decal/cleanable/traitor_rune))
+		try_complete_rune(user, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/**
+ * Attempt to draw a rune on [target_turf].
+ * Shamelessly adapted from the heretic rune drawing process.
+ *
+ * Arguments
+ * * user - the mob drawing the rune
+ * * target_turf - the place the rune's being drawn
+ */
+/obj/item/traitor_spraycan/proc/try_draw_new_rune(mob/living/user, turf/target_turf)
+	var/list/target_turfs = RANGE_TURFS(1, target_turf)
+	for(var/turf/nearby_turf as anything in target_turfs)
+		if (!isopenturf(nearby_turf) || is_type_in_typecache(nearby_turf, no_draw_turfs))
+			user.balloon_alert(user, "you need a clear 3x3 area!")
+			return
+
+	draw_rune(user, target_turf)
+
+/**
+ * Draw your stage one rune on the ground and store it.
+ *
+ * Arguments
+ * * user - the mob drawing the rune
+ * * target_turf - the place the rune's being drawn
+ */
+/obj/item/traitor_spraycan/proc/draw_rune(mob/living/user, turf/target_turf)
+	user.balloon_alert(user, "drawing outline...")
+	drawing_rune = TRUE
+	if (try_draw_step("drawing outline...", user, target_turf))
+		try_complete_rune(user, new /obj/effect/decal/cleanable/traitor_rune(target_turf))
+	drawing_rune = FALSE
```

Comment:
`drawing_rune` is checked earlier in `afterattack` so that we can exit before doing any other checks if possible, however this has identified that I am setting it here in the wrong place.